gdspy/library.py

```diff
@@ -577,19 +571,18 @@ def get_bounding_box(self):
             for reference in self.references:
                 reference_bb = reference.get_bounding_box()
                 if reference_bb is not None:
-                    bb[0, 0] = min(bb[0, 0], reference_bb[0, 0])
-                    bb[0, 1] = min(bb[0, 1], reference_bb[0, 1])
-                    bb[1, 0] = max(bb[1, 0], reference_bb[1, 0])
-                    bb[1, 1] = max(bb[1, 1], reference_bb[1, 1])
+                    all_polygons.append(reference_bb)
             if len(all_polygons) > 0:
                 all_points = numpy.concatenate(all_polygons).transpose()
                 bb[0, 0] = min(bb[0, 0], all_points[0].min())
                 bb[0, 1] = min(bb[0, 1], all_points[1].min())
                 bb[1, 0] = max(bb[1, 0], all_points[0].max())
                 bb[1, 1] = max(bb[1, 1], all_points[1].max())
+                self._bounding_box = numpy.asarray(bb)
+            else:
+                self._bounding_box = None
             self._bb_valid = True
-            self._bounding_box = bb
-        return numpy.array(self._bounding_box)
+        return self._bounding_box
```

Comment:
Here it is better to return a copy of the cached value (`numpy.array(self._bonding_box)` or `numpy.copy(self._bounding_box)`, but `numpy.copy` is less efficient), otherwise this array can be used and changed elsewhere. Then the cached value will be invalid.

Reply:
got it... wasn't following your logic. should now be fixed.